test/extended/prometheus/prometheus_builds.go

```diff
@@ -170,7 +123,75 @@ type metricTest struct {
 	greaterThan string
 }
 
-func executeOpenShiftBuild(oc *exutil.CLI, appTemplate string) {
+func runQueries(metricTests map[string][]metricTest) {
+	// expect all correct metrics within 60 seconds
+	lastErrsMap := map[string]error{}
+	for i := 0; i < 60; i++ {
+		for query, tcs := range metricTests {
+			g.By("perform prometheus metric query " + query)
+			contents, err := getBearerTokenURLViaPod(ns, execPodName, fmt.Sprintf("https://%s:%d/api/v1/query?query=%s", host, statsPort, query), bearerToken)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			correctMetrics := map[int]bool{}
+			for i, tc := range tcs {
+				result := prometheusResponse{}
+				json.Unmarshal([]byte(contents), &result)
+				metrics := result.Data.Result
+
+				for _, sample := range metrics {
+					// first see if a metric has all the label names and label values we are looking for
+					foundCorrectLabels := true
```

Comment:
If we don't have helpers to do this you should create them.  This is pretty ugly code.

Reply:
Isn't there a prometheus client library?

Reply:
re: prometheus client library, at least for my interpretation of that, there is an alpha one for remote access ....see https://github.com/prometheus/client_golang#client-for-the-prometheus-http-api and https://github.com/prometheus/client_golang/blob/master/api/prometheus/v1/api.go

but the version of prometheus vendored into origin does not have the above

there is "client_golang" stuff in the prometheus vendored in, but it is non-HTTP non-remote

unless I missed something

otherwise sure I can work up some helpers ... might do it in a non-trello card PR depending on what happens with this work